src/main/java/carpet/script/CarpetScriptServer.java

```diff
@@ -481,4 +495,25 @@ public void reAddCommands()
     {
         modules.values().forEach(host -> host.addAppCommands(s -> {}));
     }
+    
+    public static void parseFunctionClasses()
```

Comment:
What's your take on extensions and parsing their expression classes.

Reply:
Well (if I understood the question correctly), this one is to parse Carpet and base Scarpet function classes (to have all calls in a Scarpet place and not everything in CarpetServer), while extensions have `AnnotationParser#parseFunctionClass`.

If you mean the `apply(Expression expr)` method currently used, that is not really included in this system, but the old `scarpetApi` method in CarpetExtension is still there for those.

Reply:
well, its more about if that's more 'standard' way of extending scarpet, then having an automated treatment of their API call classes without explicit calls to parseFunctionClass or apply() could be preferrable. But that is outside of the scope of this PR

Reply:
Hm... I guess maybe an entrypoint could be added for Scarpet if that's what you mean? (like the startup entrypoint, but listing Scarpet function classes instead). And possibly have it be an interface with a default no-op `apply` method so extensions can add methods that are not compatible with this system if they wish.

I thought of `parseFunctionClass` as the public API for this because I considered it would be ok as a way to, in the end, communicate what are the classes that have to be checked, and because it's the same as how SettingsManager parses rules. Could definitely have some more abstraction if desired.

Reply:
Though then you still need a way to register your own values or type conversions.